pkg/jobs/canary/canary_jobs.go

```diff
@@ -253,15 +256,26 @@ func SyncCanaryJob(canary v1.Canary) error {
 		LogFail:    canary.IsTrace() || canary.IsDebug() || LogFail,
 	}
 
-	_, err := CanaryScheduler.AddJob(canary.Spec.GetSchedule(), job)
-	if err != nil {
-		return fmt.Errorf("failed to schedule canary %s/%s: %v", canary.Namespace, canary.Name, err)
+	updateTime, exists := canaryUpdateTimeCache[dbCanary.ID.String()]
+	entry := findCronEntry(canary)
+	if !exists || dbCanary.UpdatedAt.After(updateTime) || entry == nil {
+		// Remove entry if it exists
+		if entry != nil {
+			CanaryScheduler.Remove(entry.ID)
+		}
+
+		// Schedule canary for the first time
+		_, err = CanaryScheduler.AddJob(canary.Spec.GetSchedule(), job)
+		if err != nil {
+			return fmt.Errorf("failed to schedule canary %s/%s: %v", canary.Namespace, canary.Name, err)
+		}
+		logger.Infof("Scheduled %s: %s", canary, canary.Spec.GetSchedule())
+
+		canaryUpdateTimeCache[dbCanary.ID.String()] = dbCanary.UpdatedAt
 	}
-	logger.Infof("Scheduled %s: %s", canary, canary.Spec.GetSchedule())
 
-	entry = findCronEntry(canary)
-	if entry != nil && time.Until(entry.Next) < 1*time.Hour {
-		// run all regular canaries on startup
+	// Run canary on startup
```

Comment:
Run all regularly scheduled canaries on startup (<1h) and not daily/weekly schedules